modules/plottinggl/src/plotters/scatterplotgl.cpp

```diff
@@ -604,52 +491,96 @@ void ScatterPlotGL::objectPicked(PickingEvent* p) {
     const uint32_t id = static_cast<uint32_t>(p->getPickedId());
 
     // Show tooltip for current item
-    if (p->getHoverState() == PickingHoverState::Move ||
-        p->getHoverState() == PickingHoverState::Enter) {
-
+    if (properties_.tooltip_ && (p->getHoverState() == PickingHoverState::Move ||
+                                 p->getHoverState() == PickingHoverState::Enter)) {
         tooltipCallback_.invoke(p, id);
-
     } else if (p->getHoverState() == PickingHoverState::Exit) {
+        // unset tooltip at all times in case one was set prior disabling tooltips
         p->setToolTip("");
     }
 
-    if (properties_.hovering_.get()) {
+    if (properties_.showHighlighted_) {
         if (p->getHoverState() == PickingHoverState::Enter ||
             (p->getGlobalPickingId() != p->getPreviousGlobalPickingId())) {
-            highlighted_.clear();
-            highlighted_.add(id);
-            highlightChangedCallback_.invoke(highlighted_);
-            highlightDirty_ = true;
+            highlightChangedCallback_.invoke(BitSet(id));
         } else if (p->getHoverState() == PickingHoverState::Exit) {
-            highlighted_.clear();
-            highlightChangedCallback_.invoke(highlighted_);
+            highlightChangedCallback_.invoke(BitSet());
         }
     }
 
+    const double movedDistance =
```

Comment:
Could we refactor this "if moved" into some helper function at some point?

Reply:
Would be sensible, maybe even part of the PickingEvent?